sky/shell/ui/internals.h

```diff
@@ -46,6 +47,12 @@ class Internals
   mojo::ServiceProviderImpl service_provider_impl_;
   mojo::ServiceProviderPtr platform_service_provider_;
 
+  // Proxy for a ServiceProvider that exposes services from the application to
+  // the embedder.
+  mojo::ServiceProviderPtr application_service_provider_;
+  mojo::InterfaceRequest<mojo::ServiceProvider>
+      application_service_provider_request_;
```

Comment:
The names for these things get kind of crazy.  Maybe call this services_provided_by_embedder_ ?